sklearn/ensemble/tests/test_iforest.py

```diff
@@ -322,3 +322,35 @@ def test_iforest_deprecation():
     warn_msg = "'behaviour' is deprecated in 0.22 and will be removed in 0.24"
     with pytest.warns(DeprecationWarning, match=warn_msg):
         iforest.fit(iris.data)
+
+
+def test_iforest_with_uniform_data():
+    """Test whether iTrees predict inliers when using uniform data."""
+
+    # 2-d array of all 1s
+    X = np.ones((100, 10))
+    iforest = IsolationForest()
+    iforest.fit(X)
+
+    assert all(iforest.predict(X) == 1)
+    assert all(iforest.predict(np.random.randn(100, 10)) == 1)
```

Comment:
We like to have deterministic tests so you can define `rng = np.random.RandomState(0)` and use `rng` instead of `np.random` from then on